### PR TITLE
Quark SE devboard: refactor contiki-main.c

### PR DESCRIPTION
**Pull request by aguedes**
This PR refactors contiki-main.c from Quark SE devboard. No new functionality is added.

### CODE REVIEW

**Comment by jeez** [comment]
A few comments:

- please consider merging https://github.com/otcshare/contiki-x86/pull/30/commits/d9b413c1f33e099d74f77d8a5ce7a23aebdb8172 and  https://github.com/otcshare/contiki-x86/pull/28/commits/64d19e649ff1d431268ed2b184af9ef0893e4fd1

- Since you are refactoring networking code, please move it out of contiki-main.c into a separate platform file.

- I still think it is cheap to add -DQUARK_SE_DEVBOARD_NETWORK_ENABLED. It's specially useful for platform testing with no network.

**Comment by aguedes** [comment]
> please consider merging d9b413c and 64d19e6

Ok, will merge then.

> Since you are refactoring networking code, please move it out of contiki-main.c into a separate platform file.

I'm fine with them in contiki-main.c since they are just helper functions which are used during platform initialization. Actually, I think we could simply remove the print_link_local_address() function. So I prefer to leave them in contiki-main.c. If you are not OK with it, I can leave the FIXME comment in the code. Please let me know.

> I still think it is cheap to add -DQUARK_SE_DEVBOARD_NETWORK_ENABLED. It's specially useful for platform testing with no network.

If, someday, we want to implement this (support building apps with no network support), I think we can use current existing options (CONTIKI_WITH_IPV6, etc) instead of introducing another option.

**Comment by aguedes** [comment]
This PR now depends on #28. Just for reference, the first patch from this PR is "quark-se-devboard: Remove scope blocks from main()".

**Comment by jeez** [comment]
> If, someday, we want to implement this (support building apps with no network support), I think we can use current existing options (CONTIKI_WITH_IPV6, etc) instead of introducing another option.

I'm referring to the full feature. To enable network now what we do is:

  set_node_addr();
  netstack_init();
  queuebuf_init();
  setup_radio();

My suggestions here were to encapsulate all of this into a single init function that comes from a separate file. Our plat main then can just call it, i.e. network_start(); .
You could move  print_link_local_address() to the same file as well.

These functions then could be protected by a feature define like the one I mentioned on my comments, but could be added later.

This way we clean up main.c with zero overhead, since these functions will most likely be inlined anyway - (they will be called just once).

File: platform/quark-se-devboard/contiki-main.c [hunk 1]
```diff
@@ -129,17 +129,14 @@ main(void)
...
+
   setup_radio();
 
   memcpy(&uip_lladdr.addr, linkaddr_node_addr.u8,
          UIP_LLADDR_LEN > LINKADDR_SIZE ? LINKADDR_SIZE : UIP_LLADDR_LEN);
 
-  queuebuf_init();
-  NETSTACK_RDC.init();
-  NETSTACK_MAC.init();
-  NETSTACK_NETWORK.init();
```
Comment (jeez) [inline platform/quark-se-devboard/contiki-main.c]: It's worth to highlight that netstack_init() also initializes NETSTACK_LLSEC.init(), which we don't.
It's not a 1x1 change, then. I'm assuming this is not an issue, but I haven't checked it as well.

**Comment by aguedes** [comment]
> I'm referring to the full feature. To enable network now what we do is:
> 
> set_node_addr();
> netstack_init();
> queuebuf_init();
> setup_radio();
> 
> My suggestions here were to encapsulate all of these calls into a single init function that comes from a separate file. Our plat main then can just call it, i.e. network_start(); .
> You could move print_link_local_address() to the same file as well.
> 
> These functions then could be protected by a feature define like the one I mentioned on my comments, but could be added later.

Ok, I'll keep the FIXME comment then. We can revisit it if we implement that feature you mentioned.

File: platform/quark-se-devboard/contiki-main.c (hunk 1 above)
Comment (aguedes) [inline platform/quark-se-devboard/contiki-main.c]: > It's worth to highlight that netstack_init() also initializes NETSTACK_LLSEC.init(), which we don't.
> It's not a 1x1 change, then. I'm assuming this is not an issue, but I haven't checked it as well.

Yes, I was aware of this. NETSTACK_LLSEC.init() is empty.

Comment (jeez) [inline platform/quark-se-devboard/contiki-main.c]: should have mentioned on the commit msg then.

**Comment by jeez** [comment]
The suggested improvement to this PR was regardless of the feature being implemented or not.

**Pull request merged**